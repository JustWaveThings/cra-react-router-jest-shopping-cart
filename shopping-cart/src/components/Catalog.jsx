import React, { Suspense } from 'react';
import Product from './Product';
import getProducts from '../api';
import { useLoaderData, defer, Await } from 'react-router-dom';

export function loader() {
	return defer({
		product: getProducts()
	});
}

function Catalog() {
	const loaderData = useLoaderData();

	return (
		<Suspense fallback={<h2>Loading Products... </h2>}>
			<Await resolve={loaderData.product}>
				<div className="catalog--cont">
					<Product />
					<Product />
					<Product />
				</div>
			</Await>
		</Suspense>
	);
}

export default Catalog;
