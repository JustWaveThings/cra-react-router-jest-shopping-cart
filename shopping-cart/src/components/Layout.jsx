import React, { useState } from 'react';
import Header from './Header.jsx';
import { Outlet } from 'react-router-dom';
import Footer from './Footer.jsx';

function Layout() {
	const [qty, setQty] = useState(0);
	console.log(qty);
	return (
		<>
			<Header qty={qty} />
			<Outlet context={[qty, setQty]} />
			<Footer />
		</>
	);
}

export default Layout;
