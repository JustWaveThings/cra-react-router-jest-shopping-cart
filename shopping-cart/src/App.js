import React from "react";
import {
  createBrowserRouter,
  createRoutesFromElements,
  Route,
  RouterProvider,
} from "react-router-dom";
import "./index.css";
import Layout from "./components/Layout.jsx";
import Home from "./components/Home.jsx";
import Catalog from "./components/Catalog.jsx";
import Product from "./components/Product.jsx";
import Cart from "./components/Cart.jsx";
import NotFound from "./components/NotFound.jsx";

const router = createBrowserRouter(
  createRoutesFromElements(
    <Route path="/" element={<Layout />}>
      <Route index element={<Home />} />
      <Route path="catalog" element={<Catalog />} />
      <Route path="catalog/:id" element={<Product />} />
      <Route path="cart" element={<Cart />} />
      <Route path="*" element={<NotFound />} />
    </Route>
  )
);

function App() {
  return <RouterProvider router={router}></RouterProvider>;
}

export default App;