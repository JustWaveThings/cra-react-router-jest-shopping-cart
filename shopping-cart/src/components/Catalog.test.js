import React from 'react';
import { render, screen, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import Catalog from './Catalog';
import Layout from './Layout';
import App from '../App';

describe('app component', () => {
	test('snapshot test for Catalog component', () => {
		const { container } = render(
			<App>
				<Layout>
					<Catalog />
				</Layout>
			</App>
		);
		expect(container).toMatchSnapshot();
	});
});
