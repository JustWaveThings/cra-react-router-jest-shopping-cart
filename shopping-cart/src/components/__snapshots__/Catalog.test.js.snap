// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component snapshot test for Catalog component 1`] = `
<div>
  <div
    class="header--cont"
  >
    <div
      class="header--cont--left"
    >
      <a
        class="header--title"
        href="/"
      >
        the Shop
      </a>
      <div
        class="header--subtitle"
      >
         Your one-stop shop
      </div>
    </div>
    <div
      class="header--cont--center"
    >
      <a
        class="header--link--catalog"
        href="/catalog"
      >
        Catalog
      </a>
    </div>
    <div
      class="header--cont--right"
    >
      <div
        class="header--cart--count"
        data-testid="cartQty"
      >
        0
      </div>
      <a
        href="/cart"
      >
        <img
          alt="Shopping Cart Icon"
          class="header--cart"
          src="shopping-cart.svg"
        />
      </a>
    </div>
  </div>
  <div
    class="home--cont"
  >
    <div
      class="home--cta--text--button"
    >
      <h1
        class="home--title"
      >
        Chopping prices on what you want!
      </h1>
      <a
        href="/catalog"
      >
        <button
          class="home--button"
        >
          Shop Now!
        </button>
      </a>
    </div>
  </div>
  <div
    class="footer--cont"
  >
    2023 JustWaveThings
     
    <a
      href="https://github.com/JustWaveThings"
    >
      <img
        alt="github"
        height="20"
        src="github-mark-white.png"
      />
    </a>
     
  </div>
</div>
`;
